test/parallel/test-crypto-fips.js

```diff
@@ -0,0 +1,208 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var spawn = require('child_process').spawn;
+var path = require('path');
+
+if (!common.hasCrypto) {
+  console.log('1..0 # Skipped: missing crypto');
+  return;
+}
+
+const EXIT_FAILURE = -1;
+const FIPS_ENABLED = 1;
+const FIPS_DISABLED = 0;
+const FIPS_ERROR_STRING = 'Error: Cannot set FIPS mode';
+const OPTION_ERROR_STRING = 'bad option';
+const CNF_FIPS_ON = path.join(common.fixturesDir, 'openssl_fips_enabled.cnf');
+const CNF_FIPS_OFF = path.join(common.fixturesDir, 'openssl_fips_disabled.cnf');
+var num_children_spawned = 0;
+var num_children_ok = 0;
+const child_responses = {};
+
+function compiledWithFips() {
+  return process.config.variables.openssl_fips ? true : false;
+}
+
+function addToEnv(newVar, value) {
+  var envCopy = {};
+  for (const e in process.env) {
+    envCopy[e] = process.env[e];
+  }
+  envCopy[newVar] = value;
+  return envCopy;
+}
+
+function childOk(child) {
+  console.error('Child ' +  ++num_children_ok + '/' + num_children_spawned +
+      ' [pid:' + child.pid + '] OK.');
+}
+
+function testHelper(requiresFips, args, expectedOutput, cmd, env) {
+  const child = spawn(process.execPath, args.concat(['-i']), {
+    cwd: path.dirname(process.execPath),
+    env: env
+  });
+
+  console.error('Spawned child [pid:' + child.pid + '] with cmd ' +
+      cmd + ' and args \'' + args + '\'');
+  num_children_spawned++;
+
+  const response_handler = function(response, expectedOutput, expectedError) {
```

Comment:
Just curious, any point in doing it this way, instead of just `function responseHandler()`?